app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md

```diff
@@ -0,0 +1,168 @@
+    default:
+        throw new AssertionError("Should not happen");
+}
+```
+
+With [Switch Expressions](id:lang.classes-objects.switch-expression),
+the compiler can check whether all values of the enum are handled.
+```java
+DayOfWeek someDay = DayOfWeek.FRIDAY;
+
+String text = switch (someDay) {
+    case MONDAY -> "The week just started.";
+    case TUESDAY, WEDNESDAY, THURSDAY -> "We are somewhere in the middle of the week.";
+    case FRIDAY -> "The weekend is near.";
+    case SATURDAY, SUNDAY -> "Weekend";
+};
+
+System.out.println(text);
+```
+
+<a id="members">&nbsp;</a>
+## Adding members to enums
+
+Just like classes, enums can have constructors, methods and fields.
+In order to add these, it is necessary to add a `;` after the list of enum constants.
+Arguments to the constructor are passed in parenthesis after the declaration of the enum constant.
+
+```java
+public enum DayOfWeek {
+    MONDAY("MON"), TUESDAY("TUE"), WEDNESDAY("WED"), THURSDAY("THU"), FRIDAY("FRI"), SATURDAY("SAT"), SUNDAY("SUN");
+    
+    private final String abbreviation;
+    
+    DayOfWeek(String abbreviation) {
+        this.abbreviation = abbreviation;
+    }
+    
+    public String getAbbreviation() {
+        return abbreviation;
+    }
+}
+```
+
+<a id="functionality">&nbsp;</a>
+## Special methods
+
+All enums have a few methods that are added implicitely.
+
+For example, the method `name()` is present in all enum instances and can be used to get the name of the enum constant.
+Similarly, a method named `ordinal()` returns the position of the enum constant in the declaration.
+```java
+System.out.println(DayOfWeek.MONDAY.name());    // prints "MONDAY"
+System.out.println(DayOfWeek.MONDAY.ordinal()); // prints "0" because MONDAY is the first constant in the DayOfWeek enum
+```
+
+Aside from instance methods, there are also static methods added to all enums.
+The method `values()` returns an array containing all instances of the enum and the method `valueOf(String)` can be used to get a specific instance by its name.
+```java
+DayOfWeek[] days = DayOfWeek.values(); // all days of the week
+DayOfWeek monday = DayOfWeek.valueOf("MONDAY");
+```
+
+<a id="singletons">&nbsp;</a>
+## Using enums for singletons
+
+Since enums can only have a specific number of instances, it is possible to create a singleton by creating an enum with only a single enum constant.
+```java
+public enum SomeSingleton {
+    INSTANCE;
+    //fields, methods, etc.
+}
+```
+
+<a id="abstract">&nbsp;</a>
+## Abstract methods in enums
+
+Even though enums cannot be extended, they can still have `abstract` methods. In that case, an implementation must be present in each enum constant.
+```java
+enum MyEnum {
+    A() {
+        @Override
+        void doSomething() {
+            System.out.println("a");
+        }
+    },
+    B() {
+        @Override
+        void doSomething() {
+            System.out.println("b");
+        }
+    };
+    
+    abstract void doSomething();
+}
+```
```

Comment:
Some kind of summary should be good here as well. As this ends rather abruptly. Perhaps some discussion regarding when enums should and shouldn't be used? e.g. using an enum in a shared library for something that might change in the future could create a lot of headaches. Of course, if you have a better example, or way of wrapping up enums that's fine as well.